src/feature-interface.h

```diff
@@ -0,0 +1,41 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <map>
+#include <memory>
+
+
+namespace librealsense {
+
+
+// Base class for all features
+class feature_interface
+{
+public:
+    feature_interface( const std::string& id ) : _id( id )
```

Comment:
can we abstract the idea of a feature id?
`using feature_id = std::string;` in the librealsense namespace.

Reply:
And should I pass `feature_id` or `const feature_id &` as a parameter?

Reply:
feature_id